File: frontend/src/types/api/alias.ts
export type AliasInput = {
    alias_name: string;
    main_id: string;
    sub1_id: string | undefined;
    sub2_id: string | undefined;
    sub3_id: string | undefined;
    sub4_id: string | undefined;
    is_auto_registered: boolean;
}

export type AliasResponse = {
    alias_id: string;
}

File: backend/src/types/db/alias.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Serialize, Deserialize, Debug)]
pub struct Alias {
    alias_name: String,
    main_id: Uuid,
    sub1_id: Uuid,
    sub2_id: Uuid,
    sub3_id: Uuid,
    sub4_id: Uuid,
    is_auto_registered: bool,
}